### PR TITLE
Muhannad node week3

### PR DESCRIPTION
**Pull request by Muhannad1989**

### CODE REVIEW

File: week3/homework/src/actions.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    return new Promise((resolve, reject) => {
+      fs.writeFile(this._filename, JSON.stringify(data), err => {
+        if (err) {
+          reject(err);
+        }
+        resolve();
+      });
+    });
+  }
+  async update(request, response, value) {
```
Comment (eidosam) [inline week3/homework/src/actions.js]: We should be able to use this function in any environment; in web server, CLI or something else. That means it should receive the data in a parameter and returns back the results or error as fulfilled or rejected promise (exactly like what you did in `list` and `reset` functions). It is not the responsibility of this function to read from the HTTP request or send a response.

**Comment by wouterkleijn** [comment]
thanks @eidosam for the feedback. @Muhannad1989 Can you address this please?

**Comment by Muhannad1989** [comment]
Thank you Samir for your feedback , i will fix all of things you have mentioned.

File: week3/homework/src/actionsFolder/update.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    action.update(request, response, true);
+  },
+
+  setFalse: (action, request, response) => {
+    action.update(request, response, false);
+  },
+};
+
+const { setTrue, setFalse } = updatesFunction;
+module.exports = { setTrue, setFalse };
```
[the remark this answers is not in the document]
Comment (Muhannad1989) [reply week3/homework/src/actionsFolder/update.js]: yes , that possibles also 😄

File: week3/homework/src/actions.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    try {
+      let read = await this.list();
+      read = JSON.parse(read);
```
[the remark this answers is not in the document]
Comment (Muhannad1989) [reply week3/homework/src/actions.js]: i was trying to reduce as much as possible code's inside index file, i didn't want any repeated codes there, but here well.

**Comment by wouterkleijn** [comment]
Super, thanks @Muhannad1989

**Pull request closed without merging**